docs/metrics.md

```diff
@@ -0,0 +1,422 @@
+
+
+##### `language`
+
+The user's preferred language. String, gotten from [`navigator.language`](https://developer.mozilla.org/docs/Web/API/NavigatorLanguage/language).
+
+
+##### `version`
+
+The version of Firefox, gotten from [`system.version`](https://developer.mozilla.org/Add-ons/SDK/High-Level_APIs/system#version).
+
+
+##### `channel`
+
+The Firefox release channel. String, one of `nightly`, `developer`, `beta`, or `release`.
+
+
+##### `addons`
+
+A list of IDs of installed add-ons. Array.
+
+
+##### `adBlocker`
+
+Does the user have an add-on installed that is known to block ads? Boolean.
+
+
+##### `openTabs`
+
+The count of open tabs across all windows. Number.
+
+
+##### `openWindows`
+
+The number of open windows across the browser. Number.
+
+
+##### `hangs`
+
+The number of recent hangs across the browser. Number.
+
+
+#### Page data
+
+Data about the page from which the report was initiated.
+
+
+##### `timestamp`
+
+The timestamp from which the page request began. Integer.
+
+
+##### `protocol`
+
+The protocol used to retrieve the page, as reported by `window.location.protocol`. String.
+
+
+##### `protocolVersion`
+
+The version of the protocol used to retrieve the page, as reported by the `Version` header. String.
+
+
+##### `trackingProtection`
+
+Is tracking protection enabled for the page? Boolean.
+
+
+##### `timerContentLoaded`
+
+Number of milliseconds between the start of the page load and the `DOMContentLoaded` event. Number.
+
+
+##### `timerWindowLoad`
+
+Number of milliseconds between the start of the page load and the `load` event on `window`. Number.
+
+
+##### `timerFirstPaint`
+
+Number of milliseconds between the start of the page load and the first non-blank paint. Number.
+
+
+##### `timerFirstInteraction`
+
+Number of milliseconds between the start of the page load and the first of the following events:
+
+* `click`
+* `touch`
+* `scroll`
+* `keypress`
+
+Number. Will return `null` if none of these events took place.
+
+
+##### `requests`
+
+An object containing information about requests spawned by the page. Each member key indicates the type of request:
+
+* `html`: HTML files
+* `css`: Cascading style sheets
+* `js`: JavaScript
+* `xhr`: files retrieved by `XMLHttpRequest`
+* `fonts`: non-system fonts
+* `images`: image files
+* `media`: audio and video files
+* `flash`: Flash embeds
+* `ws`: WebSocket connections
+* `other`: anything not covered in the above categories
+* `all`: all requests summed.
+
+These are categorized in the same way as the [Firefox Developer Tools Network Monitor](https://developer.mozilla.org/docs/Tools/Network_Monitor).
+
+Each value of the object contains three data points:
+
+* `num`: the number of requests of that type
+* `size`: the summed filesize of requests of that type, in bytes
+* `time`: the time it took to load those filesge. Each member ke, in ty indicates the type of request:
+
+
+##### `externalRequests`
```

Comment:
"External" can be skewed by esoteric CDN setups. Maybe just count requests to domains on the [CDN list from webpagetest](https://github.com/WPO-Foundation/webpagetest/blob/master/agent/wpthook/cdn.h)